Classes/Fusion/Helper/NodeInfoHelper.php

```diff
@@ -413,7 +413,10 @@ public function uri(NodeInterface $node = null, ControllerContext $controllerCon
         }
 
         // Create an absolute URI without resolving shortcuts
-        return $this->linkingService->createNodeUri($controllerContext, $node, null, null, true, [], '', false, [], false);
+        $uri = urldecode($this->linkingService->createNodeUri($controllerContext, $node, null, null, true));
+        // @todo remove this hack when the uri is again without the [__identity] part.
+        $uri = str_replace('node[__identity]=', 'node=', $uri);
```

Comment:
I don't understand this code

Reply:
The current master was broken and the uri contains node[__indentity]= instead of node= ... this is just a QuickFix to make the UI work again. Will discuss the better solution with @bwaidelich or maybe @kdambekalns today.

Reply:
I am confused, too… this fix has nothing to do with validation and I'd never ever look for it in this PR. Which is bad, since researching the past is about a quarter of software development… This must go in a separate PR, IMHO.

Reply:
Ah now I understand you... this is just because my PR was based on the branch for the quickfix. Otherwise all test will fail because master is broken. Hope that we find a substainable solution for master soonish and then I just need to rebase the PR.

Reply:
So this comes from https://github.com/neos/neos-ui/pull/2831

Reply:
Saw that the issue was for neos 5.1. So I now changed the base branch and the discussion is resolved anyways ;)